TreeMaker/python/makeTreeFromMiniAOD_cff.py

```diff
@@ -642,7 +646,29 @@ def makeTreeFromMiniAOD(
                                          'JetsProperties:neutralHadronMultiplicity(Jets_neutralHadronMultiplicity)',
                                          'JetsProperties:photonMultiplicity(Jets_photonMultiplicity)',
                                          'JetsProperties:flavor(Jets_flavor)'])
-
+    if IncludeBEff:
+    	process.bTaggingEff        = cms.EDAnalyzer('BTaggingEffAnalyzer',
+        	JetsTag            = cms.InputTag('HTJets'),
+        	DiscriminatorTag   = cms.string('pfCombinedInclusiveSecondaryVertexV2BJetTags'),
+        	DiscriminatorValue = cms.double(0.890),
+        	PtNBins            = cms.int32(100),
+        	PtMin              = cms.double(0.),
+        	PtMax              = cms.double(1000.),
+        	EtaNBins           = cms.int32(60),
+        	EtaMin             = cms.double(-3.),
+        	EtaMax             = cms.double(3.)
+    	)
+   	process.Baseline += process.bTaggingEff
+    if AddBtagSF:
+	process.BTagScale=cms.EDProducer('BTagScale',
+		JetsTag            = cms.InputTag('HTJets'),
+		BTagEffInput   =cms.string('./data/test_run_RA2AnalysisTree.root'),
```

Comment:
Okay I will take a look at your signal skims for now, and create a version of this eff Analzyzer that runs on the treemaker skims. 
-Rishi
> On Oct 5, 2015, at 6:09 PM, Kevin Pedro <notifications@github.com> wrote:
> 
> In TreeMaker/python/makeTreeFromMiniAOD_cff.py <https://github.com/TreeMaker/TreeMaker/pull/68#discussion_r41162760>:
> 
> > +    	process.bTaggingEff        = cms.EDAnalyzer('BTaggingEffAnalyzer',
> > +        	JetsTag            = cms.InputTag('HTJets'),
> > +        	DiscriminatorTag   = cms.string('pfCombinedInclusiveSecondaryVertexV2BJetTags'),
> > +        	DiscriminatorValue = cms.double(0.890),
> > +        	PtNBins            = cms.int32(100),
> > +        	PtMin              = cms.double(0.),
> > +        	PtMax              = cms.double(1000.),
> > +        	EtaNBins           = cms.int32(60),
> > +        	EtaMin             = cms.double(-3.),
> > +        	EtaMax             = cms.double(3.)
> > +    	)
> > +   	process.Baseline += process.bTaggingEff
> > +    if AddBtagSF:
> > +	process.BTagScale=cms.EDProducer('BTagScale',
> > +		JetsTag            = cms.InputTag('HTJets'),
> > +		BTagEffInput   =cms.string('./data/test_run_RA2AnalysisTree.root'),
> To me, it makes a lot more sense to do this alongside or after my skimming step, which makes the files that get used for the datacards. This comes after TreeMaker and after splitting the signal scans into individual model points. I think we should avoid duplicating the splitting effort if possible. Is there an integration team meeting tomorrow where we could discuss this? @nhanvtran <https://github.com/nhanvtran>
> —
> Reply to this email directly or view it on GitHub <https://github.com/TreeMaker/TreeMaker/pull/68/files#r41162760>.
>